Make acknowledgements more prominently and enhance their content

HannesWell opened the pull request:
This enhances the existing [acknowledgements](https://eclipse.dev/eclipse/development/acknowledgements.html) to be more visible (I'm currently not aware of a common path through the website to it) and to enhance it's content, as currently it's only a plain text concatenating names.
With this change the acknowledgements are colocated to the news of each TLP and even linked from them. The list of contributors also becomes an actual list and each name is backed by a link to the authors GH account. I also intend to leverage https://github.com/eclipse-platform/www.eclipse.org-eclipse/pull/358 and inject the profile picture of each contributor.

Last but not least this adds a GH workflow to automate the creation and update of the contributor list in the new acknowledgements. It's currently triggered manually. Later we could also trigger this again automatically as part of preparation of the following next release (as it should be triggered in the end of a release-cycle).

Resolves
- https://gitlab.eclipse.org/eclipse-wg/ide-wg/community/-/issues/77

This is currently a draft that contains already has all the basics but requires improvements in the following points, of which some how to be discussed:
- [x] Handle old acknowledgements
  - Suggestion: Just mention in the overview page that for subsequent releases the acknowledgements are moved colocated to the news and provide a link.
- Refine logic to form the acknowledgement-entries:
  Currently the author name recorded in a git commit and the profile GH associates with it (probably by the author mail) is only used.
  - [x] filter out duplicates (different author names for same person)
  - [x] handle special cases where different GH accounts are used
  - [x] filter out bots
  - [ ] Consider co-authors?
  - [ ] Consider committers?
    - Allows to prefer the committer account instead of the author account if the same name is used. 
- Inject icons by leveraging https://github.com/eclipse-platform/www.eclipse.org-eclipse/pull/358

Impressions:
<p>
<img src="https://github.com/user-attachments/assets/b26cd15b-ca81-4edc-81f2-50bb1d6d85c1" width="300"/>
</p>
<p>
<img src="https://github.com/user-attachments/assets/c257eb04-df8e-4bc3-a4b0-270682f9ecf0" width="300"/>
</p>
<p>
<img src="https://github.com/user-attachments/assets/57ee1d03-d4b5-49c7-a83a-62e45fd0e037" width="300"/>
</p>
<p>
<img src="https://github.com/user-attachments/assets/8c394766-7fd4-4d70-bfe9-e3b41ddd9f61" width="300"/>
</p>

HannesWell commented:
The page I generated with the current (non-final) state would have the following content.

===

## Eclipse 2025-06 Acknowledgements

Last revised: June 1, 2025

The entire Eclipse Project team would like to thank the Eclipse Foundation staff for their tireless efforts, including:
Denis Roy,
Matt Ward,
Eric Poirier,
Pawel Stankiewicz,
Frederic Gurr
and Sébastien Heurtematte
for not only keeping all that infrastructure going but also constantly improving it;
Sharon Corbett for guiding us through the Eclipse legal process;
Wayne Beaton for giving us process support and improved tools to simplify and manage our development process;
and Mike Milinkovich for running the whole show.

We would also like to thank the users and adopters who support our efforts through a range of activities, including early testing, being a Friend of Eclipse, contracting special work, or outright employment.

A special thanks goes to [Holger Voormann](https://github.com/howlger/) for his Eclipse IDE promotion videos.

A special thanks to other Eclipse projects we build upon: EMF and ECF, who also provide timely updates so we can release on time.
We also thank the other Eclipse projects that make up part of the infrastructure we depend on: Tycho, Orbit, EGit, EMF, and ECF for providing fixes and steady improvements.

## Eclipse Platform

The Platform team would like to thank everyone who has helped us improve quality by testing and reporting bugs and enhancement requests.
Special thanks to all code contributors (alphabetically):
<!-- START: eclipse-platform contributors -->
- [Rahul Mohanan](https://github.com/MohananRahul)
- [dependabot[bot]](https://github.com/apps/dependabot)
- [Christoph Läubrich](https://github.com/laeubi)
- [Александър Куртаков](https://github.com/akurtakov)
- [github-actions[bot]](https://github.com/apps/github-actions)
- [Ed Merks](https://github.com/merks)
- [Eclipse Releng Bot](https://github.com/eclipse-releng-bot)
- [Hannes Wellmann](https://github.com/HannesWell)
- [Elsa Zacharia](https://github.com/elsazac)
- [danthe1st](https://github.com/danthe1st)
- [Alois Zoitl](https://github.com/azoitl)
- [azoitl](https://github.com/azoitl)
- [Andrey Loskutov](https://github.com/iloveeclipse)
- [Heiko Klare](https://github.com/HeikoKlare)
- [Віктор Дуйко](https://github.com/bravesasha)
- [Eclipse Platform Bot](https://github.com/eclipse-platform-bot)
- [Christoph Rueger](https://github.com/chrisrueger)
- [Aleksandar Kurtakov](https://github.com/akurtakov)
- [Lars Vogel](https://github.com/vogella)
- [Federico Jeanne](https://github.com/fedejeanne)
- [Suby S Surendran](https://github.com/subyssurendran666)
- [nickboldt](https://github.com/nickboldt)
- [Julien Dehaudt](https://github.com/JDESTM)
- [Jonah Graham](https://github.com/jonahgraham)
- [Alexander Kurtakov](https://github.com/akurtakov)
- [Fabrizio Iannetti](https://github.com/fabrizioiannetti)
- [Maksym Oleksiv](https://github.com/mmx85)
- [coehlrich](https://github.com/coehlrich)
- [Michael Keppler](https://github.com/Bananeweizen)
- [astrograph](https://github.com/astrograph)
- [sougandhs](https://github.com/SougandhS)
- [Holger Voormann](https://github.com/howlger)
- [Patrick Ziegler](https://github.com/ptziegler)
- [fedejeanne](https://github.com/fedejeanne)
- [John Bampton](https://github.com/jbampton)
- [Jay Arthanareeswaran](https://github.com/jarthana)
- [Sebastian Ratz](https://github.com/sratz)
- [Titouan Vervack](https://github.com/tivervac)
- [Michael Bangas](https://github.com/Michael5601)
- [raghucssit](https://github.com/raghucssit)
- [Deepika Udayagiri](https://github.com/deepika-u)
- [Andreas Koch](https://github.com/akoch-yatta)
- [Amartya Parijat](https://github.com/amartya4256)
- [Phillipus](https://github.com/Phillipus)
- [Shahzaib Ibrahim](https://github.com/ShahzaibIbrahim)
- [Simeon Andreev](https://github.com/trancexpress)
- [chris](https://github.com/chrst4)
- [Vasili Gulevich](https://github.com/basilevs)
- [braveocheretovych](https://github.com/braveocheretovych)
- [otbutz](https://github.com/otbutz)
- [konczdev](https://github.com/konczdev)
- [Manuel Killinger](https://github.com/killingerm)
- [arunjose696](https://github.com/arunjose696)
- [Oliver Lins](https://github.com/p-O-q)
- [Tobias Melcher](https://github.com/tobias-melcher)
- [Dieter Mai](https://github.com/DieterMai)
- [Jozef Tomek](https://github.com/RedeemerSK)
- [Masha](https://github.com/Marechka)
- [Matthias Becker](https://github.com/BeckerWdf)
- [Christopher Hermann](https://github.com/Christopher-Hermann)
- [Mats Lundquist](https://github.com/selundqma)
<!-- END:   eclipse-platform contributors -->

## Java Development Tools

The JDT team thanks everyone who filed good enhancement requests, helped improve quality by testing and filing bug reports, and provided answers on JDT forums/newsgroups.
Special thanks to all code contributors (alphabetically):
<!-- START: eclipse-jdt contributors -->
- [Александър Куртаков](https://github.com/akurtakov)
- [akurtakov](https://github.com/akurtakov)
- [Ed Merks](https://github.com/merks)
- [Suby S Surendran](https://github.com/subyssurendran666)
- [Jay Arthanareeswaran](https://github.com/jarthana)
- [Stephan Herrmann](https://github.com/stephan-herrmann)
- [Manoj N Palat](https://github.com/mpalat)
- [Manoj  N Palat](https://github.com/mpalat)
- [eclipse-releng-bot](https://github.com/eclipse-releng-bot)
- [Srikanth Sankaran](https://github.com/srikanth-sankaran)
- [Joerg Kubitz](https://github.com/EcljpseB0T)
- [Jesper Steen Møller](https://github.com/jespersm)
- [Jeff Johnston](https://github.com/jjohnstn)
- [dependabot[bot]](https://github.com/apps/dependabot)
- [David Thompson](https://github.com/datho7561)
- [Eclipse JDT Bot](https://github.com/eclipse-jdt-bot)
- [Rob Stryker](https://github.com/robstryker)
- [Snjeza](https://github.com/snjeza)
- [Hannes Wellmann](https://github.com/HannesWell)
- [Andrey Loskutov](https://github.com/iloveeclipse)
- [Mehmet Emin Karaman](https://github.com/mehmet-karaman)
- [Mickael Istria](https://github.com/mickaelistria)
- [Snjezana Peco](https://github.com/snjeza)
- [Noopur Gupta](https://github.com/noopur2507)
- [Zsombor](https://github.com/gzsombor)
- [Sougandh S](https://github.com/SougandhS)
- [Holger Voormann](https://github.com/howlger)
- [Zsombor Gegesy](https://github.com/gzsombor)
- [kysmith-csg](https://github.com/kysmith-csg)
- [Vasili Gulevich](https://github.com/basilevs)
- [sougandhs](https://github.com/SougandhS)
- [Michael Bangas](https://github.com/Michael5601)
- [dan1st](https://github.com/danthe1st)
- [Jakub Suliga](https://github.com/jakub-suliga)
- [Stephan Wahlbrink](https://github.com/wahlbrink)
- [Patrick Ziegler](https://github.com/ptziegler)
- [fedejeanne](https://github.com/fedejeanne)
- [braveocheretovych](https://github.com/braveocheretovych)
- [Nikifor Fedorov](https://github.com/zelenyhleb)
- [Carsten Hammer](https://github.com/carstenartur)
- [FAP](https://github.com/fapdash)
- [Shahzaib Ibrahim](https://github.com/ShahzaibIbrahim)
- [Heiko Klare](https://github.com/HeikoKlare)
- [Jozef Tomek](https://github.com/RedeemerSK)
<!-- END:   eclipse-jdt contributors -->

## Plug-in Development Environment

The Plug-in Development Environment team thanks numerous contributors who continue to improve the component every release.
Special thanks to all code contributors (alphabetically):
<!-- START: eclipse-pde contributors -->
- [Marc-Andre Laperle](https://github.com/MarkZ3)
- [Matthias Becker](https://github.com/BeckerWdf)
- [Alexander Kurtakov](https://github.com/akurtakov)
- [Lars Vogel](https://github.com/vogella)
- [Sravan Kumar Lakkimsetti](https://github.com/sravanlakkimsetti)
- [Niraj Modi](https://github.com/niraj-modi)
- [Ed Merks](https://github.com/merks)
- [Karsten Thoms](https://github.com/eCemetery)
- [Carsten Hammer](https://github.com/carstenartur)
- [Mat Booth](https://github.com/mbooth101)
- [Jonah Graham](https://github.com/jonahgraham)
- [Conrad Groth](https://github.com/ConradGroth)
- [Kit Lo](https://github.com/kitlo)
- [Andrey Loskutov](https://github.com/iloveeclipse)
- [Samantha Dawley](https://github.com/SDawley)
- [Александър Куртаков](https://github.com/akurtakov)
- [Rahul Mohanan](https://github.com/MohananRahul)
- [Deepika Udayagiri](https://github.com/deepika-u)
- [Jörg Kubitz](https://github.com/EcljpseB0T)
- [Hannes Wellmann](https://github.com/HannesWell)
- [Simeon Andreev](https://github.com/trancexpress)
- [eclipse-releng-bot](https://github.com/eclipse-releng-bot)
- [Patrick Ziegler](https://github.com/ptziegler)
- [Christoph Läubrich](https://github.com/laeubi)
- [Eclipse PDE Bot](https://github.com/eclipse-pde-bot)
- [Sebastian Ratz](https://github.com/sratz)
- [Jay Arthanareeswaran](https://github.com/jarthana)
- [Shahzaib Ibrahim](https://github.com/ShahzaibIbrahim)
- [Heiko Klare](https://github.com/HeikoKlare)
- [Michael Bangas](https://github.com/Michael5601)
- [Christoph Rueger](https://github.com/chrisrueger)
<!-- END:   eclipse-pde contributors -->

## Equinox

The Equinox team thanks all contributors who helped improve the project by filing bug reports and enhancement requests.
Special thanks to all code contributors (alphabetically): 
<!-- START: eclipse-equinox contributors -->
- [akurtakov](https://github.com/akurtakov)
- [Rahul Mohanan](https://github.com/MohananRahul)
- [Александър Куртаков](https://github.com/akurtakov)
- [Christoph Läubrich](https://github.com/laeubi)
- [Eclipse Equinox Bot](https://github.com/eclipse-equinox-bot)
- [Christoph Rueger](https://github.com/chrisrueger)
- [Eclipse Webmaster team](https://github.com/eclipsewebmaster)
- [Ed Merks](https://github.com/merks)
- [Hannes Wellmann](https://github.com/HannesWell)
- [Andrey Loskutov](https://github.com/iloveeclipse)
- [Aleksandar Kurtakov](https://github.com/akurtakov)
- [Thomas Watson](https://github.com/tjwatson)
- [JVS94](https://github.com/JVS94)
- [Michael Bangas](https://github.com/Michael5601)
<!-- END:   eclipse-equinox contributors -->

merks commented:
FYI 

it’s only discoverable via the more… page

https://eclipse.dev/eclipse/development/

MohananRahul commented:
> ## Plug-in Development Environment
> The Plug-in Development Environment team thanks numerous contributors who continue to improve the component every release. Special thanks to all code contributors (alphabetically):
> 
> * [Marc-Andre Laperle](https://github.com/MarkZ3)
> * [Matthias Becker](https://github.com/BeckerWdf)
> * [Alexander Kurtakov](https://github.com/akurtakov)
> * [Lars Vogel](https://github.com/vogella)
> * [Sravan Kumar Lakkimsetti](https://github.com/sravanlakkimsetti)
> * [Niraj Modi](https://github.com/niraj-modi)
> * [Ed Merks](https://github.com/merks)
> * [Karsten Thoms](https://github.com/eCemetery)
> * [Carsten Hammer](https://github.com/carstenartur)
> * [Mat Booth](https://github.com/mbooth101)
> * [Jonah Graham](https://github.com/jonahgraham)
> * [Conrad Groth](https://github.com/ConradGroth)
> * [Kit Lo](https://github.com/kitlo)
> * [Andrey Loskutov](https://github.com/iloveeclipse)
> * [Samantha Dawley](https://github.com/SDawley)
> * [Александър Куртаков](https://github.com/akurtakov)
> * [Rahul Mohanan](https://github.com/MohananRahul)
> * [Deepika Udayagiri](https://github.com/deepika-u)
> * [Jörg Kubitz](https://github.com/EcljpseB0T)
> * [Hannes Wellmann](https://github.com/HannesWell)
> * [Simeon Andreev](https://github.com/trancexpress)
> * [eclipse-releng-bot](https://github.com/eclipse-releng-bot)
> * [Patrick Ziegler](https://github.com/ptziegler)
> * [Christoph Läubrich](https://github.com/laeubi)
> * [Eclipse PDE Bot](https://github.com/eclipse-pde-bot)
> * [Sebastian Ratz](https://github.com/sratz)
> * [Jay Arthanareeswaran](https://github.com/jarthana)
> * [Shahzaib Ibrahim](https://github.com/ShahzaibIbrahim)
> * [Heiko Klare](https://github.com/HeikoKlare)
> * [Michael Bangas](https://github.com/Michael5601)
> * [Christoph Rueger](https://github.com/chrisrueger)
> 
> ## Equinox
> The Equinox team thanks all contributors who helped improve the project by filing bug reports and enhancement requests. Special thanks to all code contributors (alphabetically):
> 
> * [akurtakov](https://github.com/akurtakov)
> * [Rahul Mohanan](https://github.com/MohananRahul)
> * [Александър Куртаков](https://github.com/akurtakov)
> * [Christoph Läubrich](https://github.com/laeubi)
> * [Eclipse Equinox Bot](https://github.com/eclipse-equinox-bot)
> * [Christoph Rueger](https://github.com/chrisrueger)
> * [Eclipse Webmaster team](https://github.com/eclipsewebmaster)
> * [Ed Merks](https://github.com/merks)
> * [Hannes Wellmann](https://github.com/HannesWell)
> * [Andrey Loskutov](https://github.com/iloveeclipse)
> * [Aleksandar Kurtakov](https://github.com/akurtakov)
> * [Thomas Watson](https://github.com/tjwatson)
> * [JVS94](https://github.com/JVS94)
> * [Michael Bangas](https://github.com/Michael5601)

Something thing is wrong , some of contributor are not active in the list in PDE/Equinox. 
git log R4_35..master --pretty=format:"%an <%ae>" | sort | uniq

iloveeclipse commented:
> Something is wrong

Also "sorted alphabetically" seem not be correct, except it sorts by the github ID?

merks commented:
Strange indeed

This looks like the last commit that returns correct results up to ` Mon May 5 21:51:52 2025 +0200`

> $git  log I20250506-0940..master

This goes back as far as ` Tue Nov 3 23:46:18 2015 -0500`:

> $git  log I20250505-1800..master

It's a bit inexplicable.

vogella commented:
[Alois Zoitl](https://github.com/azoitl)
[azoitl](https://github.com/azoitl) is the same person. We should also remove the bots. ([Eclipse PDE Bot](https://github.com/eclipse-pde-bot)), [Eclipse JDT Bot](https://github.com/eclipse-jdt-bot) , [github-actions[bot]](https://github.com/apps/github-actions), [Eclipse Releng Bot](https://github.com/eclipse-releng-bot)

merks commented:
Of course the tag query is nicer because one can compute the tag so that nothing in the script needs to change...

HannesWell commented:
> > Something is wrong
> 
> Also "sorted alphabetically" seem not be correct, except it sorts by the github ID?

> [Alois Zoitl](https://github.com/azoitl) [azoitl](https://github.com/azoitl) is the same person. We should also remove the bots. ([Eclipse PDE Bot](https://github.com/eclipse-pde-bot)), [Eclipse JDT Bot](https://github.com/eclipse-jdt-bot) , [github-actions[bot]](https://github.com/apps/github-actions), [Eclipse Releng Bot](https://github.com/eclipse-releng-bot) and [Eclipse Equinox Bot](https://github.com/eclipse-equinox-bot)

Please see the list of active TODOs from my initial message:
- https://github.com/eclipse-platform/www.eclipse.org-eclipse/pull/361#issue-3107088023

This has not been done, but is planned to be done.




> Something is wrong , some of contributor are not active in the list in PDE/Equinox but still showing R4_35..master
> git log R4_35..master --pretty=format:"%an <%ae>" | sort | uniq

Since some spies were moved/merged from SWT to PDE in the past cycle, the git log might show more authors than actually have contributed.

HannesWell commented:
I have now enhanced the Github workflow for the generation of the acknowledgements and addressed the following points
- Avoid  duplicates (and add resolution for multiple names in the PR that updates the acknowledgements )
- Implement profile replacements for multi-profile contributors 
- Filter out bots
- Sort entries by full name (family name is difficult as it's not the last name in all cultures)
- Add reference in old acknowledgment page to new display

Content-wise respectively regarding the generated list of contributors, this is already the final state from my perspective and ready for full discussion.

I'm now looking into displaying the profile pictures next to the names like we do for the contributors in the N&N sections and will also try to further enhance the layout/arrangement of the names, as a simple vertical list needs a lot of vertical space and has a lot of white-space at the same time. I'm contemplating to use a ~three column table (but without header is difficult) or to go back to simply having all names concatenated in the text and relying on the implicit breaking of long lines. But I plan to do this after the profile-pictures are displayed so that one can assess style of the final result.

Another point that is not yet resolved for me is the handling of co-authors mentioned in the commit message
- https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors
- https://www.eclipse.org/projects/handbook/#resources-commit

In general Github recognizes them, but they seem not to be exposed by the Github-API when querying commits and I don't know a good way (yet) to find them. One solution could be, to just skip them.

HannesWell commented:
After making the pictures work, based on Ed's previous work for the N&N entries, I came up with the possible formatting options listed below. As a general remark: As this page is really about the people who contributed, I think it would be good to have their profile-pictures relatively large. At the same time it makes the text where only one or two people are mentioned, relatively hard to read and not looking well. Therefore I suggest to keep the picture smaller only for the latter case (i.e. the few people that are always mentioned).

But that being said, what's your favorite layout?
1. Unordered list
![grafik](https://github.com/user-attachments/assets/f8a761fe-d786-43be-80d9-9dd2d8fcfa56)
1. Simply listing all names in text and relying on automated line-break
![grafik](https://github.com/user-attachments/assets/67adcd59-51f7-41cb-8545-4215fbf7776d)
1. Simply listing all names in text , but with smaller profile-pictures
![grafik](https://github.com/user-attachments/assets/a28c1b1a-7d45-46de-b0e6-c57507be5fac)
1. Table (with empty headline)
![grafik](https://github.com/user-attachments/assets/c674fc1d-fd7c-4c7f-9d7d-24997525e569)
1. Table without borders (and empty invisible headline)
![grafik](https://github.com/user-attachments/assets/62d28164-3c4f-43af-b935-84da22e63369)

danthe1st commented:
I also quite like option 5 but I am not sure how mobile-friendly that option is.

Regarding relying on automatic line breaks, it might be possible to prevent a line break between the profile picture and the corresponding username (e.g. by putting both in a `<div>`?).

HannesWell commented:
> My favorite is 5 (followed by 3 and 1). It combines well-visible pictures with visual alignment (which I miss in option 3) while preserving some compactness (which I miss in option 1).

Thanks for your opinions, I agree with it and updated the generation to create a corresponding table and also updated the java-script processing and CSS to render the table as suggested.

> I also quite like option 5 but I am not sure how mobile-friendly that option is.
> 
> Regarding relying on automatic line breaks, it might be possible to prevent a line break between the profile picture and the corresponding username (e.g. by putting both in a `<div>`?).

Good point, but I'm in general not sure how mobile friendly the N&N content is, but I can check it on my phone once it's merged. But since a lot is done from a few central places we can easily enhance this later.
And since this the release is today this should be submitted basically now.


With that I consider this ready. I'll follow-up PRs to update the `applyTemplate.sh` script, add the new acknowledgements files for the 4.36 and 4.37, remove the old ones and generate the new content.

Thank you everyone for the discussion and your feedback.
